modules/nlu/src/backend/out-of-scope-featurizer.ts

```diff
@@ -26,17 +26,24 @@ function averageByPOS(utt: Utterance, posClasses: POS_SET) {
   return averageVectors(vectors)
 }
 
-export function getUtteranceFeatures(utt: Utterance): number[] {
+function countInVocab(utt: Utterance, vocab: Token2Vec) {
+  return utt.tokens.reduce((sum, t) => {
+    return sum + +!!vocab[t.toString({ lowerCase: true })]
```

Comment:
This line feels a bit weird.. + +!!

Reply:
damn I forgot the comment . notice the usage of unary operator (converting bool to number)

https://medium.com/@nikjohn/cast-to-number-in-javascript-using-the-unary-operator-f4ca67c792ce